### ISSUE TITLE
provide .bam file

### ISSUE THREAD

**Issue by MartaBenegas** [sdparekh/zUMIs#183]
Hi, I've seen that in your paper you mention that:
> Furthermore, if the user wishes to use a different mapper, it is also possible to provide zUMIs with an aligned bam file instead of the fastq file with the cDNA sequence, with the sole requirement that only one mapping position per read is reported in the bam file.

But there is not any further information. What should be the .yaml file configuration? Do we have to somehow take into account UMI/BC prior to alignment? 

Another question: I would like to run zUMIs with an [10x Chromium dataset](https://support.10xgenomics.com/single-cell-gene-expression/datasets/3.0.0/neuron_1k_v2) which has two lanes:
```
[ec2-user@ip-172-31-16-254 neuron_1k_v2_fastqs]$ ls
neuron_1k_v2_S1_L001_I1_001.fastq.gz  neuron_1k_v2_S1_L002_I1_001.fastq.gz
neuron_1k_v2_S1_L001_R1_001.fastq.gz  neuron_1k_v2_S1_L002_R1_001.fastq.gz
neuron_1k_v2_S1_L001_R2_001.fastq.gz  neuron_1k_v2_S1_L002_R2_001.fastq.gz
```
Is there any way to run zUMIs for both of them so I can obtain one single matrix?

The final one: is zUMIs compatible with SMART-seq data?

Sorry for the great number of questions, I'm trying to implement zUMIs into a pipeline so I would like to test its flexibility. Thank you very much in advance, 
Marta.

**Comment by cziegenhain** [comment]
Hi Marta,

Thanks for the continued interest in zUMIs!
1. Sorry to say but the custom bam file input support was dropped when we released zUMIs v2. 
We decided that the pipeline is so fast that it's okay to restart from fastq files.
2. When the data was sequenced across several lanes, I would recommend to run bcl2fastq with the `--no-lane-splitting` option. Alternatively, or when data is spread across several flowcells, you can simply concatenate the fastq files prior to input into zUMIs! You will then receive a single count matrix.
3.  zUMIs is compatible with all scRNA-seq protocols that I am aware of, including Smart-seq data of all versions.

Hope this helps!
best,
Christoph

**Comment by MartaBenegas** [comment]
Hi Christoph, thanks for the quick answer!
I'll continue testing it with your recommendations. I'll close this issue by now but I'll contact you if I need any further help.

Thanks for your help again, I really appreciate it,
Marta.

**Comment by cziegenhain** [comment]
Great, you're welcome!

**Comment by MartaBenegas** [comment]
While running zUMIs with the 10x dataset it raises this error:
```
Mapping...
[1] "2020-05-06 12:00:37 UTC"

EXITING because of fatal PARAMETERS error: present --sjdbOverhang=90 is not equal to the value at the genome generation step =100
SOLUTION:
```
I've tried to specify this: `additional_STAR_params: --sjdbOverhang 100` on the .yaml file before "#below, fqfilter will add a read_layout flag defining SE or PE" to overcome this issue but it seems that it is not working. Am I specifying this wrongly? Or should I redo the entire index with --sjdbOverhang 90 ?

**Comment by MartaBenegas** [comment]
I've found the option aditional_STAR_params, my bad! But now it raises this error:
```
EXITING: FATAL INPUT ERROR: duplicate parameter "sjdbOverhang" in input "Command-Line"
SOLUTION: keep only one definition of input parameters in each input source
```
So I suppose that the default value can not be overwritten and I should redo the index with a proper sdjOverhang value, right?

Sorry for the disturbances,
Marta.

**Comment by sdparekh** [comment]
Hi Marta,

as mentioned in the Wiki Usage (https://github.com/sdparekh/zUMIs/wiki/Usage), generate your STAR index without giving the sjdbOverhang option. The overhang will be automatically calculated by zUMIs on the fly according to your readlength.

Best,
Swati

**Comment by MartaBenegas** [comment]
Hi, I continue testing zUMIs!
I've already asked you if zUMIs is compatible with SMART-seq data, but I encountered that in most cases sequencing data produced by this protocol is already demultiplexed, so I have 2 fastq files (read 1 and 2) for **each cell** . Here you have one example: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107451 .
zUMIs can only process one pair of fastq files at a time, and you already recommended me to concatenate fastq from different lanes for the 10x dataset (so I did) but my doubt now is that for 10x dataset I had another fastq with sample index and reads with cell barcode but for SMART-seq2 demultiplexed datasets I don't have that information, so I'm not sure if the fastq concatenation will work in this case. Which is the recommended  procedure in these cases? (SMART-seq data demultiplexed). 

Thanks in advance,
Marta.

**Comment by cziegenhain** [comment]
Hi,

As you notice, there needs to be some sort of barcode/cell identifying sequence in the provided fastq files. 
When I encountered this issue before, I generated an arbitrary barcode sequence in fastq style format before concatenating. This is however not built into zUMIs out of the box.

Best,
Christoph

**Comment by MartaBenegas** [comment]
Okay, I'll give it a try. Just for clarity: for each cell I should make two "barcode" fastq files (one for read 1 and other for read 2) with the same number of reads as in read1/2.fastq, using the same headers as in read1/2.fastq and writing the same barcode for all reads, right? Example:
```
### reads1.fastq
>readA
AGACGGGATAGAAGTCGACCC
+
[qual]
>readB
AGGACAGGATAATATATGACAG
+
[qual]

### barcodes1.fastq
>readA
AAAAAA
+
[qual]
>readB
AAAAAA
+
[qual]
```

**Comment by cziegenhain** [comment]
Exactly you got it!
Actually you need only one barcode file for a pair of read1/2 from the same sample, that should be enough! The fastq headers should be in sync as you say, and you can set the qual to something arbitrary high so it will not get filtered (ie we accept the original sample identity as true).
Funfact: zUMIs doesn't really need the "sequence" of the barcode to be bases of ACGT, it works with any letter in my experience.

**Comment by MartaBenegas** [comment]
Got it!
Thank you very much again for your help,
Marta.

**Comment by cziegenhain** [comment]
Hi Marta,

Maybe its too late now, but since several users asked me about the problem to concatenate fastq files prior to running zUMIs, I added such functionality
https://github.com/sdparekh/zUMIs/wiki/Starting-from-demultiplexed-fastq-files

**Comment by MartaBenegas** [comment]
Hi Cristoph,
better late than never! Thanks for letting me know the new implementation :) I'll try it. 

I have another suggestion for you:
I've noticed that you return a geneInfo.tab file with information about gene id - gene name. In Seurat (a really widely used software for posterior single-cell analysis) they use the pattern "mt-" to [identify mitochondrial genes](https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html), which is present in the gene name. zUMIs uses the gene ids by default on the matrix generated and I've managed to change the names using the file you provide, but maybe it would be nice to provide an option to specify if the user wants to use gene names instead of gene ids. 

I hope you find this information useful. Thanks for your great job!
Marta.
